lib/foreman_maintain/settings.rb

```diff
@@ -0,0 +1,14 @@
+require 'yaml'
+module ForemanMaintain
+  class Settings
+    def self.load_from_file(file_dir)
+      file_path = File.join(file_dir, 'foreman_maintain.yml')
```

Comment:
For this code, it's perhaps not needed for having outside of the runner class, + it would be nice to have info about what file was used for configuration in the logger (which is quite hard doing form here, as the setup happens afterwards). What about a simpler version, directly in side the executable file

```ruby
config_file = File.expand_path('../config/foreman_maintain.yml', __FILE__)
 if File.exists?(config_file)
  config = YAML.load_file(config_file)
else
  config = {}
  default_config = true
end
ForemanMaintain.setup(config)
if default_config
  ForemanMaintain.info("Config file #{config_file} not found, using default configuration")
end
```